src/upgrade.py

```diff
@@ -189,6 +196,8 @@ def _on_upgrade_granted(self, event: UpgradeGrantedEvent) -> None:  # noqa: C901
                 memory_limit=self.charm.config.profile_limit_memory,
             )
             self.charm.unit.status = MaintenanceStatus("starting services...")
+            # stop cron daemon to be able to query `error.log`
+            set_cron_daemon("stop")
```

Comment:
this is inhibiting `error.log` rotation, so the file can be read to capture the specific error that indicates data directory incompatibility